### PR TITLE
ASTRACTL-17973 :  Added code changes to sync ldap users and groups

### PR DESCRIPTION
**Pull request by sakdhawan**
1. Added code changes to sync ldap users and groups
2. Added test cases for functions added in library

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -3895,3 +3882,96 @@ func (client *gocloak) DeleteClientScopesScopeMappingsClientRoles(ctx context.Co
...
+
+	if err := checkForError(resp, err, errMessage); err != nil {
+		return nil, err
+	}
+	return result, nil
+}
+
+// AddGroupMapper will add ldap mapper
+func (client *gocloak) AddGroupMapper(ctx context.Context, accessToken, realm string, reqBody LdapGroupMapper) error {
+	const errMessage = "failed to create ldap mapper"
```
Comment (bp974) [inline client.go:3925]: Should this be `failed to create group mapper`?

Comment (sakdhawan) [reply client.go:3925]: no it should not fail

Comment (satyabhp) [reply client.go:3925]: @sakdhawan Asked is here to change the error message from "failed to create ldap mapper" to "failed to create group mapper"

File: models.go [hunk 2]
```diff
@@ -1324,6 +1324,50 @@ type CheckLdapConnection struct {
...
+
+// LdapGroupMapper represents request body for ldap group mapper
+type LdapGroupMapper struct {
+	Config       *map[string][]string `json:"config"`
+	Name         *string              `json:"name"`
+	ProviderId   *string              `json:"providerId"`
+	ProviderType *string              `json:"providerType"`
+	ParentId     *string              `json:"parentId"`
+}
+
```
Comment (satyabhp) [inline models.go:1339]: Can we create a separate struct as ProviderDetail with below field
         Name         *string              `json:"name"`
	ProviderId   *string              `json:"providerId"`
	ProviderType *string              `json:"providerType"`
	ParentId     *string              `json:"parentId"`

and use in both  LdapMapperDetail and LdapGroupMapper struct, I could see duplicate fields in both LdapMapperDetail and LdapMapperDetail

Comment (sakdhawan) [reply models.go:1339]: We can create a diff struct issue is not that but when we use that struct inside both other structures it changes the request and response format which we cannot do, so they will have to be duplicated only

File: client.go (hunk 1 above)
Comment (sakdhawan) [reply client.go]: made the changes

**Review by ghakazian — APPROVED** [review]

**Pull request merged**